File: sdks/node/src/domain/plugin/service.ts
import type { ICyanPlugin } from '../core/cyan_script.js';
import type { PluginInput } from './input.js';
import type { PluginOutput } from './output.js';

class PluginService {
  readonly #plugin: ICyanPlugin;

  constructor(plugin: ICyanPlugin) {
    this.#plugin = plugin;
  }

  async plug(input: PluginInput): Promise<PluginOutput> {
    const { directory, config } = input;
    return await this.#plugin.plugin({
      directory,
      config,
    });
  }
}

export { PluginService };

File: sdks/python/domain/plugin/service.py
from domain.core.cyan_script import ICyanPlugin
from domain.core.cyan_script_model import CyanPluginInput
from domain.plugin.input import PluginInput
from domain.plugin.output import PluginOutput


class PluginService:
    def __init__(self, plugin: ICyanPlugin):
        self._plugin = plugin  # Leading underscore indicates a "private" attribute

    async def plug(self, i: PluginInput) -> PluginOutput:
        directory = i.directory
        config = i.config
        return await self._plugin.plugin(CyanPluginInput(directory=directory, config=config))
